### DISCUSSION TITLE
act() method warning in tests

### DISCUSSION THREAD

**Discussion by StefanZivkovic** [testing-library/jest-dom#330] [Q&A]
In my test every call of fireevent.change or fireevent.blur i have wrapped with 
```
await act(async () => {
   fireEvent.blur(streetAddress);
});
```

Anyhow, i keep getting warning after test execution:

```Warning: An update to Formik inside a test was not wrapped in act(...).
  When testing, code that causes React state updates should be wrapped into act(...):
    act(() => {
      /* fire events that update state /  
    }); 
    / assert on the output */```
    

Is someone willing to explain?

**Comment by gnapse** [comment]
This is not a jest-dom issue. jest-dom is not tied to react or has anything to do with it. The `fireEvent` utility is from `@testing-library/react` and ultimately from `@testing-library/dom`. You should really request help about this in those libraries' help/support channels.

That being said, it would be helpful to have a way to reproduce your problem for others to better be able to help you. It would go a long way if you extracted a minimal app example with tests reproducing the problem. For example, in codesandbox.
